lib/apiservers/engine/backends/system.go

```diff
@@ -288,6 +288,10 @@ func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.A
 		registryAddress = registryAddress + "/v2/"
 	}
 
+	if !strings.HasPrefix(registryAddress, "https://") && !strings.HasPrefix(registryAddress, "http://") {
+		registryAddress = "https://" + registryAddress
```

Comment:
I think that you should look for [letters]:// prefix rather than specific protocols, it should be more transparent for the future.